Seed sample data using fixtures, factories, and Faker for development and testing

Harendra-Singh-Rathore opened the pull request:
## Summary
Added seed sample data using fixtures, factories, and Faker to support testing and development. This includes dynamically generated data for Users, Movies, Genres, Ratings, and Comments. The goal is to provide realistic sample data for local development and testing environments.

## Context
We need a robust set of sample data to populate the database and facilitate testing across different parts of the application. By utilizing:

- Faker for generating dynamic, realistic data.
- Factories to define reusable model blueprints.
- Fixtures for data seeding in development and test environments.

This ensures a smoother development experience by providing mock data for models and allowing us to easily run tests without manually creating data.

## Artifacts
- `test/fixtures/admins.yml`
- `test/fixtures/users.yml`
- `test/fixtures/movies.yml`
- `test/fixtures/genres.yml`
- `test/fixtures/movie_genres.yml`
- `test/fixtures/comments.yml`
- `test/fixtures/ratings.yml`
- `test/factories/movies.rb`
- `db/seeds/movies_only.rb`


## Impact
- [x] Low (UI / small change)  
- [ ] Medium (DB query, new model, background job)  
- [ ] High (affects login, payments, or core flows)  

### Screenshots
N/A (no UI changes)

### Related Issue
Closes #4

Harendra-Singh-Rathore committed:
feat: add seed sample data using fixtures, factories, and Faker

- Integrated Faker for generating dynamic movie test data
- Added factory definitions for movie model
- Created fixture files to seed sample data

mandalorian99 commented on Gemfile:
```diff
@@ -44,9 +44,14 @@ gem "bootsnap", require: false
 # gem "image_processing", "~> 1.2"
 
+# For generating fake data
+gem 'faker'
```
add this to development group

Harendra-Singh-Rathore committed:
feat: Set up database seed script with fixtures and FactoryBot

- Add database cleanup for all models
- Load fixtures in specific order for dependencies
- Include FactoryBot to generate additional movie records
- Ensure proper data seeding sequence

Harendra-Singh-Rathore commented:
Closes #4
